### ISSUE TITLE
@Index not created in DB

### ISSUE THREAD

**Issue by jmai00** [typeorm/typeorm#242]
I have AbstractEntity that has columns with Index, but the index are not being created in the concrete entity table.

```typescript
import {Column, PrimaryGeneratedColumn, AbstractTable, Index} from "../src/index";

@AbstractTable()
export class Asset {

    @PrimaryGeneratedColumn()
    id: number;

    @Column({name: "number", type: "string", nullable: true})
    @Index()
    number: string;
}
```

```typescript
import {User} from "./User";
import {Asset} from "./Asset";
import {JoinColumn, OneToOne, Table} from "../src/index";

@Table("Workstation")
export class Workstation extends Asset {

    @OneToOne((type) => User, {cascadeInsert: true, nullable: false})
    @JoinColumn({name: "userId"})
    user: User;
}
```
```typescript
import {Table, Column, PrimaryGeneratedColumn, Index} from "../src/index";

@Table()
@Index("index_first_last_name", (user: User) => [user.firstName, user.lastName], {unique: true})
export class User {

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    firstName: string;

    @Column()
    lastName: string;
}
```

```typescript
require("reflect-metadata");
import {createConnection, ConnectionOptions, DriverOptions, Connection} from "./src/index";
import {Person} from "./entities/Person";
import {Department} from "./entities/Department";
import {Employee} from "./entities/Employee";
import {Address} from "./entities/Address";

createConnection(<ConnectionOptions>{
    driver: {
        type: "mysql",
        host: "localhost",
        port: 3306,
        username: "root",
        password: "root",
        database: "test_dev",
    },
    entities: [
        Asset, Workstation, User,
    ],
    dropSchemaOnConnection: true,
    autoSchemaSync: true,
    logging: {
        logQueries: true,
    },
}).then( async (connection: Connection) => {
});
```

**Comment by pleerock** [comment]
That was a bug. Fixed and released in `0.0.8-alpha.5`.

**Comment by jmai00** [comment]
This your bug fixed only handled at the Table level. The example I gave is at the attribute class member level.

**Comment by pleerock** [comment]
fixed and released in `0.0.9-alpha.1`

**Comment by jmai00** [comment]
Thank you very much!!!
